<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Analyzing transcriptional regulation with NGS (UNAM LCG BEII)</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <meta name="viewport" content="width=device-width">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>Analyzing transcriptional regulation with NGS (UNAM LCG BE-II)</h1>
        
        <p>Course taught by Jacques van Helden to 2nd year students of the Licenciatura en Ciencias Genomicas de la Universidad Nacional Autonoma de Mexico</p>

        <p class="view"><a href="https://github.com/jvanheld/LCG_Bioinfo-II">View the Project on GitHub <small>jvanheld/LCG_Bioinfo-II</small></a></p>


        <ul>
          <li><a href="https://github.com/jvanheld/LCG_Bioinfo-II/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/jvanheld/LCG_Bioinfo-II/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/jvanheld/LCG_Bioinfo-II">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
        <h3>
<a id="scope-of-the-course" class="anchor" href="#scope-of-the-course" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Scope of the course</h3>

<p>This Web site contains the teaching material for a 1-week course taught by Jacques van Helden to 2nd year's students of the Licenciatura en Ciencias Genomicas (LCG) of the Universidad Nacional Autonoma de Mexico (UNAM).</p>

<p>The goal of the course is to introduce bioinformatics and statistical methods to analyse gene regulation based on Next Generation Sequencing data. </p>

<h3>
<a id="keywords" class="anchor" href="#keywords" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Keywords</h3>

<ul>
<li>Next Generation Sequencing (NGS)</li>
<li>Regulatory Sequence Analysis Tools (RSAT)</li>
<li>Bioinformatics</li>
<li>Statistics</li>
<li>Genomics</li>
<li>ChIP-seq</li>
<!--<li>RNA-seq</li>-->
</ul>

<h3>
<a id="authors-and-contributors" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Authors and Contributors</h3>

<p>Jacques van Helden
<br>Institut Fran&ccedil;ais de Bioinformatique (IFB).
<br>Aix-Marseille Universit√© (AMU).
<br>Lab. Theory and Approaches of Genome Complexity (TAGC)
<br>Web:  <a href="http://jacques.van-helden.perso.luminy.univ-amu.fr/">http://jacques.van-helden.perso.luminy.univ-amu.fr/</a>
<br>Email: <a href="mailto:Jacques.van-Helden@univ-amu.fr">Jacques.van-Helden@univ-amu.fr</a>



<hr>
<h3>
<a id="teaching-material" class="anchor" href="#teaching-material" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Teaching material (2017-01)</h3>

<table>
  <tr><th>Day 1: from TF binding sites to motifs and back</th></tr>
  <tr valign='top'>
    <td>Transcriptional regulation</td>
    <td>[<a target='_blank' href='http://rsa-tools.github.io/course/pdf_files/01.2_regulatory_sequences_intro.pdf'>pdf</a>]</td>
  </tr>
  
  
  <tr>
    <td>Position-specific scoring matrices (PSSM)</td>
    <td>[<a target='_blank' href='http://rsa-tools.github.io/course/pdf_files/01.4.PSSM_theory.pdf'>pdf</a>]</td>
  </tr>

  <tr>
    <td>Sequence models (Bernoulli, Markov chains)</td>
    <td>[<a target='_blank' href='http://rsa-tools.github.io/course/pdf_files/01.3.sequence_models.pdf'>pdf</a>]</td>
  </tr>

  
  <tr>
    <td>Matrix-based pattern matching</td>
    <td>[<a target='_blank' href='http://rsa-tools.github.io/course/pdf_files/02.3.pattern_matching_matrices.pdf'>pdf</a>]</td>
  </tr>
  
  <tr>
    <td>Home work: scaning sequences with a reference motif</td>
  </tr>

 <tr><th>Day 2: motif discovery</th></tr>
  
  <tr valign='top'>
    <td>Debriefing of first day home work</td>
  </tr>
 
  <tr valign='top'>
    <td>Intro du motif discovery</td>
    <td>[<a target='_blank' href='http://rsa-tools.github.io/course/pdf_files/03.1.pattern_discovery_intro.pdf'>pdf</a>]</td>
    </tr>
    
    <tr>
      <td>String-based pattern discovery</td>
      <td>[<a target='_blank' href='http://rsa-tools.github.io/course/pdf_files/03.2.1.pattern_discovery_strings.pdf'>pdf</a>]</td>
  </tr>
  


</table> <!--
  <tr><th>Day 2</th></tr>


  <tr valign='top'>
    <td>Introduction to Next Generation Sequencing (NGS)</td>
    <td>[<a target='_blank' href='slides/01_NGS_intro_DP.pdf'>pdf</a>]</td>
  </tr>
  
  <tr valign='top'>
    <td>ChIP-seq technologies and applications</td>
    <td>[<a target='_blank' href='slides/ChIP-seq_intro.pdf'>pdf</a>]</td>
  </tr>

  <tr valign='top'>
    <td>ChIP-seq analysis 1: read-mapping and quality control</td>
    <td>[<a target='_blank' href='slides/ChIP-SEQ_analysis_with_Galaxy.pdf'>pdf</a>]</td>
  </tr>

  <tr valign='top'>
    <td>ChIP-seq analysis 2: peak calling</td>
    <td>[<a target='_blank' href='slides/ChIP-SEQ_analysis_with_Galaxy.pdf'>pdf</a>]</td>
  </tr>

  <tr><th>Day 3</th></tr>
  
  <tr>
    <td>Peak-calling statistics with R: exercise</td>
    <td>
    [<a target='_blank' href='practicals/peak-calling_statistics/peak-calling_statistics_exercise.html'>html</a>]
    [<a target='_blank' href='practicals/02_peak-calling/peak-calling_stats_practical.pdf'>pdf</a>]
    [<a target='_blank' href='practicals/02_peak-calling/peak-calling_stats_practical.R'>R code</a>]
    <small>[<a target='_blank' href='practicals/peak-calling_statistics/peak-calling_statistics_exercise.Rmd'>Rmd</a>]</small>
  </tr>

  <tr>
    <td>Peak-calling statistics with R: beginining of the solutions</td>
    <td>
    [<a target='_blank' href='practicals/peak-calling_statistics/peak-calling_statistics_beginning.html'>html</a>]
     [<a target='_blank' href='practicals/02_peak-calling/peak-calling_stats_practical.pdf'>pdf</a>]
    [<a target='_blank' href='practicals/02_peak-calling/peak-calling_stats_practical.R'>R code</a>]
    <small>[<a target='_blank' href='practicals/peak-calling_statistics/peak-calling_statistics_beginning.Rmd'>Rmd</a>]</small>
  </tr>


<tr>
    <td>Peak-calling statistics with R: tutorial</td>
    <td>
    [<a target='_blank' href='practicals/02_peak-calling/peak-calling_stats_practical.html'>html</a>]
    [<a target='_blank' href='practicals/02_peak-calling/peak-calling_stats_practical.pdf'>pdf</a>]
    [<a target='_blank' href='practicals/02_peak-calling/peak-calling_stats_practical.R'>R code</a>]
    <small>[<a target='_blank' href='practicals/02_peak-calling/peak-calling_stats_practical.Rmd'>Rmd</a>]</small>
  </tr>

  <tr>
    <td>Peak-calling statistics with R: commented report</td>
    <td>
    [<a target='_blank' href='practicals/02_peak-calling/peak-calling_report.html'>html</a>]
    [<a target='_blank' href='practicals/02_peak-calling/peak-calling_report.pdf'>pdf</a>]
    [<a target='_blank' href='practicals/02_peak-calling/peak-calling_report.R'>R code</a>]
    <small>[<a target='_blank' href='practicals/02_peak-calling/peak-calling_report.Rmd'>Rmd</a>]</small>
    </td>
  </tr>
  

  <tr><th>Day 4</th></tr>
  
  <tr valign='top'>
    <td>ChIP-seq analysis 3: motif analysis</td>
    <td>[<a target='_blank' href='slides/ChIP-seq_peak-motifs.pdf'>pdf</a>]</td>
  </tr>

  <tr valign="top">
    <td>Pracical: ENCODE ChIP-seq peaks</td>
    <td>
      [<a target="_blank" href="practicals/encode_chip-seq/encode_ChIP-seq_analysis.html">html</a>]
      [<a target="_blank" href="practicals/encode_chip-seq/encode_ChIP-seq_analysis.Rmd">Rmd</a>]
    </td>
  </tr>
  <tr><th>Day 5</th></tr>

  <tr>
    <td>Peak-calling statistics with R: solutions (2017)</td>
    <td>
    [<a target='_blank' href='practicals/peak-calling_statistics/peak-calling_statistics_solutions_2017.html'>html</a>]
    [<a target='_blank' href='practicals/02_peak-calling/peak-calling_stats_practical.pdf'>pdf</a>]
   [<a target='_blank' href='practicals/02_peak-calling/peak-calling_stats_practical.R'>R code</a>]
    <small>[<a target='_blank' href='practicals/peak-calling_statistics/peak-calling_statistics_solutions_2017.Rmd'>Rmd</a>]</small>
  </tr>
  
  <tr>
    <td>Report template for the personal work</td>
    <td>[<a href="practicals/encode_chip-seq/YOUR_NAME_ENCODE_ChIP-seq_motifs.Rmd">Rmd</a>]</td>
  </tr>


</table>

-->

<hr><h3><a id="resources" class="anchor" href="#resources" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Resources</h3>

<table>
  <tr>
    <td>RSAT</td>
    <td>Regulatory sequence analysis tools <br><a target='_blank' href='http://rsat.eu/'>http://rsat.eu/</a></td>
  </tr>
  
  <tr>

    <td>RSAT course</td>
    <td>Teaching material about the analysis of regulatory sequences 
      <br><a target='_blank' href='http://rsa-tools.github.io/course/'>http://rsa-tools.github.io/course/</a></td>
  </tr>
  
  <tr>
    <td>ChIP-seq tutorial</td>
    <td>Tutorial on chip-seq analysis written by Morgane Thomas-Chollier, from which I used the FNR dataset used in this course.
    <br><a target='_blank' href='http://www.biologie.ens.fr/~mthomas/other/chip-seq-training/'>http://www.biologie.ens.fr/~mthomas/other/chip-seq-training/</a></td>
  </tr>
  
  <tr>
    <td>
      Stats4bioinfo
    </td>
    <td>
      Supporting site for an introductory course about statistics for bioingormatics.
      <br><a target='_blank' href='http://pedagogix-tagc.univ-mrs.fr/courses/statistics_bioinformatics/'>http://pedagogix-tagc.univ-mrs.fr/courses/statistics_bioinformatics/</a>
    </td>
  </tr>
  
  <tr>
    <td>AVIESAN NGS ChIP-seq course</td>
    <td><a target='_blank' href='http://jvanheld.github.io/stats_avec_RStudio_EBA/'>http://jvanheld.github.io/EBA15_stats_RStudio/</a></td>
  </tr>
  
  <tr>
    <td>
      <a target='_blank' href='http://microbes.ucsc.edu/'>Microbes UCSC</a>
    </td>
    <td>
      A mirror of the UCSC genome browser dedicated to microbial genomes.
  </tr>
  
</table>



<!--
<table>
  <tr valign='top'>
    <td>Day&nbsp;1</td>
    <td>
      <ol>
        <li><a target='_blank' href="slides/01_Introduction_content.html">Introduction and course content</a></li>
        <li><b>Practical:</b> <a target='_blank' href='http://jvanheld.github.io/cisreg_course/from_sites_to_matrices/from_sites_to_matrices.html'>From TF binding sites to consensuses and matrices</a></li>
        <li><b>Practical:</b> <a target='_blank' href='http://jvanheld.github.io/cisreg_course/sequence_scanning/promoter_scanning_ecoli.html'>Scanning sequences to predict TF binding sites</a></li>
      </ol>
    </td>
  </tr>
  
  <tr valign='top'>
    <td>Day&nbsp;2</td>
    <td>
      <ol>
        <li><a target='_blank' href='https://docs.google.com/spreadsheets/d/1jePHHAdBwVcyBtKPeHFx0xIzFRdZNcrc0tsbgoiBc0o/edit?usp=sharing'>Summaries of the sequence scanning results</a></li>
        
        <li>Comparison and discussion of student results for the sequence scanning tutorial (one regulon per student).</li>
        
        <li><a target='_blank' href='http://rsa-tools.github.io/course/pdf_files/03.2.1.pattern_discovery_strings.pdf'>String-based pattern discovery</a></li>
        [<a target='_blank' href='http://cursos.lcg.unam.mx/claroline/document/goto/?url=%2FJacques_van-Helden_BEII_2016%2FRSAT_protocols_for_students_only%2Fchapter_RSAT_plant_ChIP-seq_motifs_submitted_2016-02-08.pdf&cidReq=BE2_2016'>pdf</a>]</li>
        
        <li><b>Practical</b>: RSAT::Plants: motif discovery within <b>clusters of upstream sequences</b> in plant genomes
        [<a target='_blank' href='http://cursos.lcg.unam.mx/claroline/document/goto/?url=%2FJacques_van-Helden_BEII_2016%2FRSAT_protocols_for_students_only%2Fsequence-cluster-protocol.html&cidReq=BE2_2016'>html</a>]
        [<a target='_blank' href='http://cursos.lcg.unam.mx/claroline/document/goto/?url=%2FJacques_van-Helden_BEII_2016%2FRSAT_protocols_for_students_only%2Fsequence-cluster-protocol.pdf&cidReq=BE2_2016'>pdf</a>]
        <i>Revised thanks to the feedback of LCG_BEII students, 2016-02-09</i></li>

      </ol>
    </td>
  </tr>
  
  <tr valign='top'>
    <td>Day&nbsp;3</td>
    <td>
      <ol>
        <li><a target='_blank' href='slides/chip-seq_adapted_from_Carl_Herrmann_AVIESAN.pdf'>ChIP-seq analysis</a></li>
        
        <li><a target='_blank' href='http://jvanheld.github.io/EBA15_stats_RStudio/practicals/01_first_steps_with_R/01_first_steps_with_R.html'>First steps with R tutorial (EBA15)</a></li>
        
        <li><a target='_blank' href='http://jvanheld.github.io/EBA15_stats_RStudio/practicals/02_peak-calling/peak-calling_stats_practical.html#(1)'>Basic peak-calling statistics (EBA2015)</a></li>
        <ul>
          <li><b>Datasets:</b> local copy of the peak-calling statistics dataset: <a target='_blank' href='http://cursos.lcg.unam.mx/claroline/document/goto/?url=%2FJacques_van-Helden_BEII_2016%2Fpeak-calling_statistics_data%2Fpeak-calling_statistics.zip&cidReq=BE2_2016'>peak-calling_statistics.zip</a></li>
        
          <li><b>Solutions:</b> peak-calling statistics
            [<a target='_blank' href='practicals/day3_peak-calling_stats/peak-calling_statistics_practicals.html'>html</a>]
            [<a target='_blank' href='practicals/day3_peak-calling_stats/peak-calling_statistics_practicals.Rmd'>Rmd</a>]</li>
          </ul>

      </ol>
    </td>
  </tr>
  
  <tr valign='top'>
    <td>Day&nbsp;4</td>
    <td>
      <ol>
        <li><a target='_blank' href='slides/12.1.ChIP-seq_motif_tools.pdf'>A motivated view on ChIP-seq peaks</a></li>
        <li>Peak-calling statistics (continued)</li>
     </ol>
    </td>
  </tr>
  
  <tr valign='top'>
    <td>Day&nbsp;5</td>
    <td>
      <ol>
        <li><b>Practical</b>: RSAT::Plants: motif discovery in <b>ChIP-seq peaks</b> of plant genomes
        [<a target='_blank' href='http://cursos.lcg.unam.mx/claroline/document/goto/?url=%2FJacques_van-Helden_BEII_2016%2FRSAT_protocols_for_students_only%2Fchapter_RSAT_plant_ChIP-seq_motifs.html&cidReq=BE2_2016'>html</a>]</li>
        <li><a target='_blank' href='practicals/day5_homework/homework_peak-calling_choices-2016-02-11.html'><b>Homework</b></a>
        <li>Report template
        [<a target='_blank' href='practicals/day5_homework/NAME1_NAME2_chip-seq_evaluation_report.Rmd' download>Rmd</a>]
        [<a target='_blank' href='practicals/day5_homework/NAME1_NAME2_chip-seq_evaluation_report.docx' download>docx</a>]
        </li>
       </ol>
    </td>
  </tr>

</table>

-->

<h3>
<a id="support-or-contact" class="anchor" href="#support-or-contact" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Support or Contact</h3>

<p><a href="mailto:Jacques.van-Helden@univ-amu.fr">Jacques.van-Helden@univ-amu.fr</a></p>

<hr>
      </section>
      <footer>
        <p>This project is maintained by <a href="https://github.com/jvanheld">jvanheld</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
    
  </body>
</html>
