---
title: "Practicals - basic peak-calling statistics"
author: "Jacques van Helden"
date: '`r Sys.Date()`'
output:
  html_document:
    fig_caption: yes
    highlight: zenburn
    theme: cerulean
    toc: yes
    toc_depth: 3
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
  word_document: default
---


## Introduction


The goal of this tutorial is to ....

```{r}
## We will print "Hello"
print("Hello")
```


## Data loading

We will first load the datasets.

We will load bedgraph-formatted files indicating the counts of reads per 200bp window along *E.coli* genome. 


```{r path to the course}
## Define the base URL of the course and store it in a variable
url.course <- "http://jvanheld.github.io/stats_avec_RStudio_EBA/"

## Define the fill path of the data directory, by concatenating the course URL and all the successive folders
url.data <- file.path(url.course, "practicals", "02_peak-calling", "data")

```


```{r loading the ChIP-seq counts}

## Define URL of the ChIP file
chip.bedg.file <- file.path(url.data, "FNR_200bp.bedg")
print(chip.bedg.file)

## Load the file content in an R data.frame
chip.bedg <- read.table(chip.bedg.file)

## Set column names
names(chip.bedg) <- c("chrom", "start", "end","counts")


## Check the dimensions of the bedgraph for the ChIP-seq
dim(chip.bedg)
summary(chip.bedg)
```

We loaded a tab-delimited file with the counts of reads per window in the FNR ChIP-seq sample. The table contains `r nrow(chip.bedg)` rows, each row corresponding to a genomic window. 


```{r loading the input counts}
## Define URL of the input file
input.bedg.file <- file.path(url.data, "input_200bp.bedg")

## Load the file content in an R data.frame
input.bedg <- read.table(input.bedg.file)

## Set column names
names(input.bedg) <- c("chrom", "start", "end","counts")

```

## Histograms of counts per window

We will first compare the distribution of counts per windows in ChIP-seq and input, respectively. 

```{r count_histograms}

par(mfrow=c(3,1))

## Histogram of counts per window for the ChIP-seq sample
hist(chip.bedg$counts,
     main="ChIP-seq reads per window",
     ylab="Number of windows",
     xlab="Counts",
     col="darkgreen", border = "darkgreen",
     breaks= 1000)


hist(chip.bedg$counts,
     main="ChIP-seq reads per window (truncated)",
     ylab="Number of windows",
     xlab="Counts", xlim=c(0,800),
     col="darkgreen", border = "darkgreen",
     breaks= 2000)

hist(input.bedg$counts,
     main="Input reads per window (truncated)",
     ylab="Number of windows",
     xlab="Counts", xlim=c(0,800),
     col="red", border = "red",
     breaks= 2000)
par(mfrow=c(1,1))

```

## Compute log2(counts)

```{r}

count.stats <- data.frame(chip.counts= chip.bedg$counts,
                          input.counts = input.bedg$counts)


epsilon <- 1/16
count.stats$chip.epsilon <- count.stats$chip.counts
count.stats$chip.epsilon[count.stats$chip.counts==0] <- epsilon
count.stats$input.epsilon <- count.stats$input.counts
count.stats$input.epsilon[count.stats$input.counts==0] <- epsilon

count.stats$chip.log2.counts <- log2(count.stats$chip.epsilon)
count.stats$input.log2.counts <- log2(count.stats$input.epsilon)

summary(count.stats)

```

```{r log2_count_histograms}

par(mfrow=c(2,1))

## Histogram of counts per window for the ChIP-seq sample
hist(count.stats$input.log2.counts,
     main="ChIP-seq reads per window",
     ylab="Number of windows",
     xlab="Counts", xlim=c(-4, 20),
     col="darkgreen", border = "darkgreen",
     breaks= 2000)

hist(count.stats$chip.log2.counts,
     main="Input reads per window (truncated)",
     ylab="Number of windows",
     xlab="Counts",  xlim=c(-4, 20), 
     col="red", border = "red",
     breaks= 2000)
par(mfrow=c(1,1))

```


## Compute the mean position per window

```{r}

## Compute the middle position for each window
chip.bedg$midpos <- (chip.bedg$start + chip.bedg$end)/2

## Check the result
head(chip.bedg)
```

## Plot the coverage profiles of the FNR ChIP-seq and input samples

We will plot on top of each other the coverage profiles of the ChIP-seq (FNR) and input samples. 

Figure 1 shows the coverage profiles of the ChIP-seq (top panel, darkgreen) and input (bottom panel, red) samples. 

Note the huge difference of scale: the input has a maximum of `r max(input.bedg$counts)` counts per bin, whereas the highest ChIP-seq bin contains `r max(chip.bedg$counts)` counts.

```{r covergage_profiles, fig.width=7, fig.height=12, fig.cap="**Figure 1.** Coverage profiles of the ChIP-seq (dark green) and input (red) samples. "}

par(mfrow=c(2,1))

plot(x = chip.bedg$midpos, 
     y = chip.bedg$counts,
     col="darkgreen", 
     main="FNR ChIP",
     xlab="Genomic position (200bp windows)", 
     ylab="Read counts",
     type="h")

## Compute middle positions per window
input.bedg$midpos <- (input.bedg$start + input.bedg$end)/2

plot(input.bedg[, c("midpos", "counts")], 
     col="red", 
     main="Background (genomic input)",
     xlab="Genomic position (200bp windows)", 
     ylab= "Read counts",
     type="h")


par(mfrow=c(1,1))

```


## Normalization


We would like to draw a differential profile, by substracting the input from the CHIP-seq. However, this must be done on the normalized counts, in order to compensate the differences of library sizes. 

### Library sizes

```{r}
## Compute the library sizes and their ratio
input.libsize <- sum(input.bedg$counts)
chip.libsize <- sum(chip.bedg$counts)
libsize.ratio <- chip.libsize / input.libsize
```


The library sizes are very different between the ChIP-seq sample (`r chip.libsize` reads) and the input (`r input.libsize`). We thus need to normalize the libraries in order to ensure a consistent scaling. 

A naive way to normalize would be to multiply the input library by a scaling factor $s$, in order to bring it to the same size as the input. 

$$s = N_{\text{chip}} / N_{\text{input}}$$


### Median-based scaling

The median is a robust estimate of the central tendency, since its value only depends on the centralmost part of hte distribution. We can use it as scaling factor to compensate the differences in library sizes.

```{r median_noramlization}

input.median <- median(count.stats$input.counts)
chip.median <- median(count.stats$chip.counts)
median.ratio <- chip.median / input.median

count.stats$input.median.norm <- count.stats$input.counts * print(median.ratio)
```




### Compute cumulative distribution of read counts per window

```{r}
## Beware, we rank the bins by increasing counts to get the empty bins first and the densest bins last
count.stats$chip.rank <- rank(count.stats$chip.counts)
count.stats$input.rank <- rank(count.stats$input.counts)

## Compare the ranks
#plot(count.stats$chip.rank, count.stats$input.rank)

## Compute cumulative counts per rank
chip.cumsum <- cumsum (sort(count.stats$chip.counts))
input.cumsum <- cumsum (sort(count.stats$input.counts))
chip.cumpercent <- chip.cumsum / max(chip.cumsum)*100
input.cumpercent <- input.cumsum / max(input.cumsum)*100
nbins <- length(chip.cumsum)



## Plot the cumulative distribution of read counts in input and ChIP samples, resp
par(mfrow=c(2,1))

# plot(input.cumsum, chip.cumsum)
plot(1:nbins, input.cumpercent, type="l", 
     ylab="Percent of cumulated reads",
     xlab= "Bin rank",
     col="red", lwd=2)
grid(col="blue")
lines(1:nbins, chip.cumpercent, type="l", col="darkgreen", lwd=2)
abline(a=0, b=100/nbins, lty="dashed")

## Compute the difference between cumulative distributions
diff.cumpercent <- input.cumpercent- chip.cumpercent
plot(1:nbins, diff.cumpercent, type="l", 
     ylab="Differential cumulative distribution",
     xlab= "Bin rank",
     col="blue", lwd=2)
grid(col="blue")
abline(h=0, lty="dashed")

rank.cutoff <- which.max(diff.cumpercent)
arrows(x0=rank.cutoff, 4, rank.cutoff, 7.5)

nbins
max(input.cumsum)
max(chip.cumsum)
input.cumcounts.at.cutoff <- input.cumsum[rank.cutoff]
chip.cumcounts.at.cutoff <- chip.cumsum[rank.cutoff]

cumsum.norm.ratio <- chip.cumcounts.at.cutoff  /  input.cumcounts.at.cutoff

## Choose the cumsum ratio as normalizing

norm.factor <- cumsum.norm.ratio
count.stats$input.counts.norm <- count.stats$input.counts * norm.factor

```


## Draw an XY plot with ChIP-seq versus input 


```{r}
par(mfrow=c(1,1))
plot(x= count.stats$input.counts, y= count.stats$chip.counts)

plot(x= count.stats$input.counts.norm, 
     xlab="Bin-wise estimation of exp. counts",
     ylab="ChIP-seq counts per bin",
     y= count.stats$chip.counts, col="#888888",
     xlim=c(0,500), ylim=c(0,500))
grid(col="blue")
abline(a=0, b=1, col="black", lty="dashed")


```

```{r}
## Same plot with log axes
plot(x= count.stats$input.counts.norm, 
     xlab="Bin-wise estimation of exp. counts",
     ylab="ChIP-seq counts per bin",
     y= count.stats$chip.counts, col="#888888", log="xy")
grid(col="blue")
abline(a=0, b=1, col="black", lty="dashed")

```


## Count ratios

The most intuitive (and spontanous) way of comparing tow numbers is to comupte their ratio. However, the ratio can be problematic because it will emphasize small-number fluctuations.

We can test this visually, by plotting the **count ratio** as a function of ...

```{r}
count.stats$count.ratio <- count.stats$chip.counts / count.stats$input.counts.norm
plot(count.stats$input.counts.norm, count.stats$count.ratio)

```


## Log2-ratio

```{r}

count.stats$log2.ratio <- log2(count.stats$count.ratio)
plot(count.stats$input.counts.norm, count.stats$log2.ratio)

```

## Compute the p-value

In a first time, we will naively consider that the normalized input counts per bin can serve as estimate of the expected counts in the same bins. 


```{r}
count.stats$pval.binwise <- ppois(
  q = count.stats$chip.counts - 1, 
  lambda = count.stats$input.counts.norm, 
  lower.tail=FALSE)
count.stats$eval.binwise <- count.stats$pval.binwise * nbins

plot(
  count.stats$input.counts.norm,
  -log10(count.stats$eval.binwise),
  col="#666666"
     )
grid(col="blue")

```

```{r volcano plot}
alpha=0.05

## Volcano plot
plot(
  count.stats$log2.ratio, 
  -log10(count.stats$pval.binwise),
  xlab="log2FC",
  ylab="-log10(p-value)",
  cex=0.5,
  col="#888888")
abline(h=0, col="black")
abline(v=0, col="black")
grid(col="blue")
abline(h=-log10(alpha), col="red")
```



