---
title: "Practicals - basic peak-calling statistics"
author: "Jacques van Helden"
date: '`r Sys.Date()`'
output:
  html_document:
    fig_caption: yes
    highlight: zenburn
    theme: cerulean
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
  word_document: default
---


# Introduction


The goal of this tutorial is to explore ChIP-seq data and to evaluate various statistics and graphical modes in order to detect peaks, i.e. regions with a significantly higher number of reads than what would be expected based on some background model. 

We will use a dataset from the bacteria *Escherichia coli K12*. Since bacteria have a relatively small genome compared to metazoans, the ChIP-seq dataset achieves a very high coverage, which gives us particularly interesting conditions to assess the validity of our prior assumptions.

********************
# Tasks

1. **Load** the ChIP-seq coverage profile and the genomic input. 

2. Plot the **coverage profiles**. 

3. Compute and represent graphically the **distributions of counts per bin**.

4. Compute and draw the **Lorenz curves** for the ChIP-seq data and for the genomic input^[Tip: you can use the R `cumsum()` function. ]. 

5. **Normalize** the counts per library (ChIP-seq, input) in order to compensate for the different sequencing depths. For this, you need to define a normalizing factor, which will multiply the counts in each sample. As normalizing factor, you can use the *libsum* (sum of read counts per sequencing library), but you might also think about more robust normalization factor, for example the median count, or the  $90^{th}$ percentile of the counts per bins.

6. Compute various scores that might be used to **compare the counts** per bin between ChIP-ped and input datasets. For each of the metrics, draw a plot to show the values of the comparison metrics. We might for example think about the following metrics:

    - $d = f_{i,C} - f_{i,G}$, the **difference** between normalized frequencies of the chip ($C$) and genomic input ($G$);
    - $r = \frac{f_{i,C}}{f_{i,G}}$, ChIP_seq / input ratio between normalized counts;
    - $log2FC = log_2(\frac{f_{i,C}}{f_{i,G}})$, the log-fold-change, i.e. the log2(ratio) between normalized frequencies;
    - $LLR = f_{i,G} \cdot ln(\frac{f_{i,C}}{f_{i,G}})$, i.e. background frequency multiplied by the log2(ratio);
    - ...
    
7. Based on the Poisson distribution, compute a **p-value** per bin, as well as an **adjusted p-value** (corrected for multiple testing). 

8. Draw a **volcano plot** with the log2-fold-change as measure of the effect size (abscissa) and the $-log_{10}(p_{adj})$ on the ordinate.

9. Select the **significant bins**.

10. **View the results in IGV** and compare the peaks identified with this rudimentary strategy with those returned by a real peak-caller. 

****************************************************************

# Data loading

We will first load the datasets.

We will load bedgraph-formatted files indicating the counts of reads per 200bp window along *E.coli* genome. 


```{r path to the course}
## Define the base URL of the course and store it in a variable
url.course <- "http://jvanheld.github.io/stats_avec_RStudio_EBA/"

## Define the fill path of the data directory, by concatenating the course URL and all the successive folders
url.data <- file.path(url.course, "practicals", "02_peak-calling", "data")

```


```{r loading the ChIP-seq counts}

## Define URL of the ChIP file
chip.bedg.file <- file.path(url.data, "FNR_200bp.bedg")
print(chip.bedg.file)

## Load the file content in an R data.frame
chip.bedg <- read.table(chip.bedg.file)

## Set column names
names(chip.bedg) <- c("chrom", "start", "end","counts")


## Check the dimensions of the bedgraph for the ChIP-seq
dim(chip.bedg)
summary(chip.bedg)
```

We loaded a tab-delimited file with the counts of reads per window in the FNR ChIP-seq sample. The table contains `r nrow(chip.bedg)` rows, each row corresponding to a genomic window. 


```{r loading the input counts}
## Define URL of the input file
input.bedg.file <- file.path(url.data, "input_200bp.bedg")

## Load the file content in an R data.frame
input.bedg <- read.table(input.bedg.file)

## Set column names
names(input.bedg) <- c("chrom", "start", "end","counts")

```


# Results


... TO DO ...

