---
title: "Discovering a motif discovering appproach"
author: "Jacques van Helden"
date: '`r Sys.Date()`'
output:
  html_document:
    code_folding: hide
    self_contained: no
    fig_caption: yes
    highlight: zenburn
    theme: cerulean
    toc: yes
    toc_depth: 3
    toc_float: yes
  revealjs::revealjs_presentation:
    theme: night
    transition: none
    self_contained: true
    css: ../../slides.css
  slidy_presentation:
    smart: no
    slide_level: 2
    self_contained: yes
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    highlight: tango
    incremental: no
    keep_md: yes
    smaller: yes
    theme: cerulean
    toc: yes
    widescreen: yes
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
  ioslides_presentation:
    slide_level: 2
    self_contained: no
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    smaller: yes
    toc: yes
    widescreen: yes
  beamer_presentation:
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    incremental: no
    keep_tex: no
    slide_level: 2
    theme: Montpellier
    toc: yes
font-import: http://fonts.googleapis.com/css?family=Risque
subtitle: LCG_BEII 2019
font-family: Garamond
transition: linear
---

```{r include=FALSE, echo=FALSE, eval=TRUE}
library(knitr)
library(kableExtra)
# library(formattable)

options(width = 300)
# options(encoding = 'UTF-8')
knitr::opts_chunk$set(
  fig.width = 7, fig.height = 5, 
  fig.path = 'figures/R_intro_',
  fig.align = "center", 
  size = "tiny", 
  echo = TRUE, eval = TRUE, 
  warning = FALSE, message = FALSE, 
  results = TRUE, comment = "")

options(scipen = 12) ## Max number of digits for non-scientific notation
# knitr::asis_output("\\footnotesize")

```

## Goal of the exercise

The goal of this exercise is to get an intuition of a motif discovery approach relying on the detection of over-represented oligonucleotides. 

Our approach will be pragmatic. 

We retrieved the upstream non-coding sequences of the genes involved in methionine biosynthesis and sulfur assimilation, and counted the occurrences of each hexanucleotide. 

We also computed 

- the relative frequencies (occurrences of each oligo / sum of all oligo occurrences) in the sequence of interest (the promoters of methionine-associated genes)
- the relative frequencies of ach hexanucleotide in the whole set of yeast promoters.

We would like to know if some  6nt are over-reprsented in promoters of methionine-associated genes relative to the occurrences that would be expected from a random selection of yeast promoters.

## Create a workspace for this practical

- In your home directory, create a work directory for this practical (for example `~/LCG_BEII/practical_motif_discovery/`).

```{r}
workdir <- "~/LCG_BEII/practical_motif_discovery"
dir.create(workdir, showWarnings = FALSE, recursive = TRUE)
setwd(workdir)
```


## Loading the data table


1. Download the oligonucleotide count table. 
[Scerevisiae_MET-genes_oligos-6nt-2str-noov_occ-freq.tsv](http://jvanheld.github.io/LCG_BEII/practicals/motif_discovery/data/Scerevisiae_MET-genes_oligos-6nt-2str-noov_occ-freq.tsv)

```{r download_oligo_counts}
oligo.url <- "http://jvanheld.github.io/LCG_BEII/practicals/motif_discovery/data/Scerevisiae_MET-genes_oligos-6nt-2str-noov_occ-freq.tsv"
oligo.file <- basename(oligo.url) ## Suppress the URL path and keep only the file name for local storage
download.file(oligo.url, destfile = oligo.file)
```


2. In **R**, open a new script or R markdown file. 

3. Load the data table, print the 5 top rows and the 5 bottom rows.

```{r load_oligos}

oligo.table <- read.delim(oligo.file, header = 1, row.names = 1)
# View(oligo.table)

head(oligo.table, n = 5)
tail(oligo.table, n = 5)
```

4. Draw an histogram of the observed occurrences and evaluate the spread of counts. 

5. Draw a scatter plot comparing the observed and expected occurrences for each hexanucleotide. 

6. Compute the log-ratio of observed / expected occurrences, and draw a scatter plot with this log-ratio (Y) as a function of the expected occurrences (X). 

$$lr = log(x/<X>)$$

7. Compute the log-likelihood ratio ($llr$), defined below, and draw a scatter plot with this $llr$ as a function of the expected occurrences. 

$$llr = f \cdot log(x/<X>)$$


8. Use the binomial distribution to compute the P-value of the observed occurrences. 

$$P = T(X \ge x)$$

9. Draw an histogram with the P-values of all hexanucleotides, with 20 bins.

10. Draw a scatter plot with the P-value (Y) as a function of the log-ratio (X). 

11. Compute the E-value, and the significance. 

$$E = P \cdot N$$
$$sig = -log_{10}(E)$$

12. Draw a **Volcano plot**, with the significance as a function of the log-ratio. 





