<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2020-02-04" />

<title>Solutions - Scanning non-coding sequences with a TFBM</title>

<script src="motif_search_RegulonDB_solutions_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="motif_search_RegulonDB_solutions_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="motif_search_RegulonDB_solutions_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="motif_search_RegulonDB_solutions_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="motif_search_RegulonDB_solutions_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="motif_search_RegulonDB_solutions_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="motif_search_RegulonDB_solutions_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="motif_search_RegulonDB_solutions_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="motif_search_RegulonDB_solutions_files/navigation-1.1/tabsets.js"></script>
<script src="motif_search_RegulonDB_solutions_files/navigation-1.1/codefolding.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Solutions - Scanning non-coding sequences with a TFBM</h1>
<h3 class="subtitle">LCG_BEII 2020</h3>
<h4 class="author">Jacques van Helden</h4>
<h4 class="date">2020-02-04</h4>

</div>


<div id="solutions-to-the-exercises" class="section level2">
<h2>Solutions to the exercises</h2>
<p>In this file we provide the solutions of the practical <strong>Scanning non-coding sequences with a TFBM</strong> with command-line use of the <em>RSAT</em> software suite.</p>
</div>
<div id="collective-table-for-the-2020-practical" class="section level2">
<h2>Collective table for the 2020 practical</h2>
<p>Students will store their results in a shared spreadsheet, which will be used to compare their results and get a broader landscape from the comparison of the results obtained with different transcription factors.</p>
<ul>
<li>Folder: <a href="https://tinyurl.com/lcg-beii-2020" class="uri">https://tinyurl.com/lcg-beii-2020</a></li>
<li>Motif scanning exercise:</li>
</ul>
<p>In your computer, create a folder to store the results of this practical, for example : <code>$HOME/LCG_BEII_practicals/</code> (you can change the path and name according to your own organisation of folders).</p>
</div>
<div id="choosing-a-tf-on-regulondb" class="section level2">
<h2>Choosing a TF on RegulonDB</h2>
<p>For this exercise, I chose the transcription facot LexA. I define this in an environment variable</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="bu">export</span> <span class="va">FACTOR=</span>LexA</a></code></pre></div>
<ul>
<li><p>Open a connection to RegulonDB <a href="http://regulondb.ccg.unam.mx/" class="uri">http://regulondb.ccg.unam.mx/</a></p></li>
<li><p>Click on the link <a href="http://regulondb.ccg.unam.mx/search?term=Regulon&amp;organism=ECK12&amp;type=All#">regulon list</a>. This opens a table with all the regulons.</p></li>
<li><p>Choose a TF of interest and open its record</p></li>
<li><p>Fill up the details of the collective exploration table (<a href="https://tinyurl.com/lcg-beii-19" class="uri">https://tinyurl.com/lcg-beii-19</a>).</p></li>
<li><p>Save a fasta file with the sequences of the known binding sites for your TF (tip: click on the bug “+” button in the header of the binding site section)</p></li>
<li><p>Save in a text file the matrix associated to your factor.</p></li>
</ul>
</div>
<div id="computing-the-degenerate-consensus-from-the-reference-matrix" class="section level2">
<h2>Computing the degenerate consensus from the reference matrix</h2>
<ul>
<li><p>Connect RSAT server: <a href="http://rsat.eu/" class="uri">http://rsat.eu/</a></p></li>
<li><p>Choose the bacterial server</p></li>
<li><p>Use <strong>convert-matrix</strong> to compute frequencies, weights, parameters and display a logo of your matrix.</p></li>
<li><p>In the result, get the degenerated consensus and save it to a separate text file.</p></li>
</ul>
</div>
<div id="getting-all-upstream-promoter-sequences-of-e.coli" class="section level2">
<h2>Getting all upstream (“promoter”) sequences of <em>E.coli</em></h2>
<ul>
<li><p>Open the tool <strong>retrieve-seq</strong></p></li>
<li><p>Select organism <em>Escherichia coli K12</em> (top : type simply K12 in the organism query box)</p></li>
<li>Set all parameters to get the non-coding sequences located upstream of all genes with a maximal distance of 400 bp from the gene start</li>
<li><p>Copy the URL of the result file and save it in a text file (we will use it several times below)</p></li>
</ul>
</div>
<div id="coverage-of-the-annotated-binding-sites-by-the-reference-motif" class="section level2">
<h2>Coverage of the annotated binding sites by the reference motif</h2>
<ul>
<li><p>Use <strong>dna-pattern</strong> to scan the annotated binding sites (extracted from RegulonDB) with the degenerate consensus.</p></li>
<li><p>Use <strong>matrix-scan</strong> to scan the same sites with the RegulonDB matrix</p></li>
<li><p>Compare the coverage rate of the two motifs</p></li>
</ul>
</div>
<div id="binding-site-prediction-in-all-promoters" class="section level2">
<h2>Binding site prediction in all promoters</h2>
<ul>
<li><p>Use the same tools (dna-pattern and matrix-scan) to predict binding sites in all the promoters of E.coli.</p></li>
<li><p>For <strong>matrix-scan</strong>, run the analysis with a threshold of p-value of either 0.001 or 0.0001.</p></li>
<li><p>Compare the number of matches obtained in these respective searches.</p></li>
<li><p>With the respective p-values used for the scanning, how many matches would you expect by chance ?</p></li>
</ul>
</div>
<div id="negative-control-1-scan-artificial-sequences-with-your-motif" class="section level2">
<h2>Negative control 1: scan artificial sequences with your motif</h2>
<ul>
<li>RSAT random sequences</li>
</ul>
</div>
<div id="negative-control-2-permute-the-columns-of-the-matrix" class="section level2">
<h2>Negative control 2: permute the columns of the matrix</h2>
<ul>
<li><p>Use the tool <strong>permute-matrix</strong> in order to generate 10 randomized copies of the motif</p></li>
<li><p>Send these randomiazed matrices to <strong>convert-matrix</strong> and check their logo.</p></li>
<li><p>Run the same analyses as above with the randomized matrix</p></li>
<li><p>Compare the number of sites obtained between the RegulonDB matrix and the randomized matrix derived from it.</p></li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
