---
title: 'Homework: peak-calling choices'
author: "Jacques van Helden"
date: '`r Sys.Date()`'
output:
  html_document:
    fig_caption: yes
    highlight: zenburn
    theme: cerulean
    toc: yes
    toc_depth: 3
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
  word_document: default
---


## Introduction

We saw that there are many peak callers, and that the number of peaks varies depending on some parametric choices. How to evaluate the choices ?

A variety of approaches could be envisaged to deal with this question:


1. **Significance**


    - Compare discovered motif significance in actual peak with random genomic regions of the same sizes.

2. **Consistency**

    - Compare peak sets returned by different peak callers
    Motif analysis
    
3. **Motif enrichment**

    - Compare the enrichment of the different peak sets for the reference motif
    - If no available reference motif: compare relative enrichment for discovered motifs

4. **Peak annotation**

    - Check the fraction of peaks in genomic regions compatible with regulation (promoters, introns)
    - Eukaryotes: compare peak sets with histone marks associated with enhancers
    - Bacteria/Fungi: evaluate the phylogenetic conservation of predicted binding sites (phylogenetic footprints)

5. Comparison with a **golden standard dataset**

    - If a reference collection of sites is available, evaluate the overlap. Note: we can only evaluate the sensitivity, not the specificity since all site databases are incomplete.

## Goal of the homework

Each student should choose one of the two chips-seq datasets seen during the course.

1. Escherichia coli FNR 
2. Arabidopsis thaliaba MYB3R

The goal of the homework is to combine a subset of the approaches listed above in order to  compare two peak-calling results, and evaluate their respective relevance.  

The comparison can take various modalities:

- comparison between two programs (e.g. SWEMBL vs MACS2) 
- comparison between peaksets returned by the same program with different parameter values
- analysis of the "depth" of the peak set: evalution of the relevance of the peaks depending on their scores (available in the bed files).

Note: no one is expected to perform all the comparisons between all peaksets with each of the approaches listed above. Rather, I ask each one to select a few peak sets and one or two approaches. 


## Datasets

The datasets are available on the server of the training (tepeu) in the root of the rsat user account.

- ~rsat/FNR_Ecoli_chip-seq
- ~rsat/Athaliana-Myb

Each of these folders contains several Gigabytes of files organized in folders and sub-folders. Since you do not need the complete file sructure for your work, we indicate hereby a command that allows you to get the list of all the peak coordinates (.bed files) and peak sequences (.fasta files).


### Subfolder organization

InFor the E.coli project, the peak calling results are in the folder ~rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/


The folder name indicate the peak-caller (macs2 or SWEMBL) and the parameter used to tune the stringency of the peak-calling. 

For macs2, the q-value is the estiamtin of false discovery rate, the highest stringency is thus associated to the smaller q-values. 

| Nb of peaks | File path |
|-------------|-----------|
|354|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/macs2-qval0.0001_peaks/GSM1010219_vs_GSM1010224_bowtie2_macs2-qval0.0001_peaks.bed
|390|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/macs2-qval0.001_peaks/GSM1010219_vs_GSM1010224_bowtie2_macs2-qval0.001_peaks.bed
|454|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/macs2-qval0.01_peaks/GSM1010219_vs_GSM1010224_bowtie2_macs2-qval0.01_peaks.bed
|534|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/macs2-qval0.05_peaks/GSM1010219_vs_GSM1010224_bowtie2_macs2-qval0.05_peaks.bed
|596|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/macs2-qval0.1_peaks/GSM1010219_vs_GSM1010224_bowtie2_macs2-qval0.1_peaks.bed


For SWEMBL, the parameter "R" (background) strongly affects the number of peaks returned. 

| Nb of peaks | File path |
|-------------|-----------|
|2939|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/swembl-R0.05/GSM1010219_vs_GSM1010224_bowtie2_swembl-R0.05.bed|
|1610|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/swembl-R0.06/GSM1010219_vs_GSM1010224_bowtie2_swembl-R0.06.bed|
|834|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/swembl-R0.07/GSM1010219_vs_GSM1010224_bowtie2_swembl-R0.07.bed|
|306|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/swembl-R0.1/GSM1010219_vs_GSM1010224_bowtie2_swembl-R0.1.bed|
|167|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/swembl-R0.2/GSM1010219_vs_GSM1010224_bowtie2_swembl-R0.2.bed|
|105|/home/rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224/swembl-R0.5/GSM1010219_vs_GSM1010224_bowtie2_swembl-R0.5.bed|


### E.coli FNR chip-seq

```
## Peak coordinate files
find ~rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224 -name '*.bed'

## Peak sequence files corresponding to the bed files
find ~rsat/FNR_Ecoli_chip-seq/results/GSM1010219_vs_GSM1010224 -name '*.bed' | perl -pe 's|\.bed$|.fasta|' | xargs ls -1
```

### A.thaliana MYB3R chip-seq

```
## Peak coordinate files
find ~rsat/Athaliana-Myb/GSM1482283_vs_GSM1482284/ -name '*.bed'

## Peak sequence files corresponding to the bed files
## (ignore the bed files with no associated fasta)
find ~rsat/Athaliana-Myb/GSM1482283_vs_GSM1482284/ -name '*.bed' | perl -pe 's|\.bed$|.fasta|' | xargs ls -1
```

## Suggested tools


| Tool | URL | Usage |
|------|-----|-------|
| bedtools | <http://bedtools.readthedocs.org/> | extremely useful and efficient suite to  analyse various types of genomic regions |
| peak-motifs | <http://rsat.eu/> | Motif discovery in ChIP-seq peaks |
| compare-features<br>convert-features |  <http://rsat.eu/> | Comparisons between genomic features (peak sets, genome annotations, reference binding sites, ...) |
| matrix-quality | <http://rsat.eu/> | enrichment of sequences (e.g. peak set) for user-provided motifs (e.g. reference motif for the ChIP-ped factor) |

* * * * * * 

## Structure of the report

- space limitation: max 3 pages of text. Non-included: figures, tables and supplemntary material

- language: Espa√±ol o Ingles

- report should be structured as a mini-article (see sections below)

We provide a template report indicating the expected content and organization of the report. 

- [Template report (Rmd)](NAME1_NAME2_chip-seq_evaluation_report.Rmd)
- [Template report (docx)](NAME1_NAME2_chip-seq_evaluation_report.docx)

## Format of the report to be delivered
 
- The report should be delivered in a readable format (preferably HTML, but could also be pdf or docx) as well as the original Rmd file that served to generate it. 

- The files should be deposited in the tareas area of the LCG server. 

- Optionally you can also indicate in the report the URL of your github repository if you choose to define one (put it as supplementary material).



* * * * * * *



