---
title: "Practical: analysis of ChIP-seq peaks"
author: "Jacques van Helden"
date: '`r Sys.Date()`'
output:
  html_document:
    self_contained: no
    fig_caption: yes
    highlight: zenburn
    theme: cerulean
    toc: yes
    toc_depth: 3
    toc_float: yes
  ioslides_presentation:
    slide_level: 2
    self_contained: no
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    smaller: yes
    toc: yes
    widescreen: yes
  revealjs::revealjs_presentation:
    theme: night
    transition: none
    self_contained: true
    css: ../../slides.css
  slidy_presentation:
    smart: no
    slide_level: 2
    self_contained: yes
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    highlight: tango
    incremental: no
    keep_md: yes
    smaller: yes
    theme: cerulean
    toc: yes
    widescreen: yes
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
  beamer_presentation:
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    incremental: no
    keep_tex: no
    slide_level: 2
    theme: Montpellier
    toc: yes
font-import: http://fonts.googleapis.com/css?family=Risque
subtitle: LCG_BEII 2019
font-family: Garamond
transition: linear
---

```{r include=FALSE, echo=FALSE, eval=TRUE}
library(knitr)
library(kableExtra)
# library(formattable)

options(width = 300)
# options(encoding = 'UTF-8')
knitr::opts_chunk$set(
  fig.width = 7, fig.height = 5, 
  fig.path = 'figures/R_intro_',
  fig.align = "center", 
  size = "tiny", 
  echo = TRUE, eval = TRUE, 
  warning = FALSE, message = FALSE, 
  results = FALSE, comment = "")

options(scipen = 12) ## Max number of digits for non-scientific notation
# knitr::asis_output("\\footnotesize")

```


## Introduction

In this practrical, we will combine various bioinformatics resources to extract information from ChIP-seq experiments, and evaluate the quality of the peaks.

## Resources


| Resource Name | URL |
|-------------------------|--------------------------------------|
| ReMap | A database of ChIP-seq peaks (<http://remap.cisreg.eu/>) |
| Jaspar | A database of eukaryote TF binding motifs, mainly built from CHIP-seq peaks (<http://jaspar.genereg.net/>) |
| Hocomoco | A database of Human TF binding motifs, mainly built from CHIP-seq peaks (<http://hocomoco11.autosome.ru/>) |
| RSAT| Regulatory Sequence Analysis Tools <http://rsat.eu/> |
| RSAT Metazoa | <http://metazoa.rsat.eu/> |
| UCSC NGS file formats | <https://genome.ucsc.edu/FAQ/FAQformat.html> | 

## Creating a workspace for this practical

- Create a new folder to store the results of this practical (e.g.  `~/LCG_BEII/chip-seq_practical/`)
- Set this folder as your workiing direct

## Getting peaks from ReMap

ReMap (<http://remap.cisreg.eu/>) is a database of TF binding regions based on an extensive re-analysis of published ChIP-seq data for human transcription factors. ory (`setwd()`)

- Open a connection to ReMap
- Explore the interface
- Choose a transcription factor and a tissue / cell type of interest (for this tutorial I will use Sox2 in Embryonic stem cells, ESC) 
- Get the peak coordinates for your TF in the selected tissue / cell type

## Getting motifs from reference databases

- Open a connection to Jaspar

    - Explore the database functionalities

    - Find the transcription factor binding motifs for your TF of interest. Download the corresponding matrix in your working directory.

- Do the same with the [Hocomoco](http://hocomoco11.autosome.ru/) database. 

    - Note the classification of transcription factors on the home page. You can browse it to observe TF families, defined based on their binding domains. 
    - Find the transcripiton factor of interest and save its binding motif in your work directory. 
    

## Discovering motifs with RSAT peak-motifs


- Open a connection to [RSAT Metazoa](http://metazoa.rsat.eu/)

- Use the tool **fetch-sequences** to get the sequences of your peaks. 

    **BEWARE:** the peaks of all ChIP-seq datasets were computed by Ballester's team based on the version hg38 of Human genome

- Send the results to **peak-motifs** 

    - as reference databases, select both Jaspar non-redundant vertebrates and Hocomoco
    - leave all other parameters unchanged an run the analyses

- Interpret the results: 

    - Did you obtain significant results? How significant ?
    - If so, were they characterized by over-representation (*oligo-analysis*), positional bias (*position-analysis*) or both?
    - Did some of the discovered motifs match the reference motifs for your TF in Jaspar and/or Hocomoco ?
    - Did you discover other motifs
    - 



## Negative control

- Use **random genome fragments** to pick up random regions having the same sizes as your peaks

- Run the same analyses as above with peak-motifs

- Compare the results of this negative control with those obtained above with your peaks, and interpret them. 



## Motif encrihment analysis

- On the RSAT server, open the tool**matrix-quality**

- Submit your peak sequences

- Enter the JASPAR and Hocomoco reference motifs

- Run the analyses

- Interpret the results

    - Do you observe an enrichment of your peaks for Hocomoco and/or Jaspar reference motifs?
    



```bash
awk '{len=$3 - $2; sum += len;  print len"\t"sum}' practicals/chip-seq_analysis/data/ReMap_GSE69479.SOX2.ESC_peaks.bed

```


## Instruction for the report

1. Volume: < 5 pages of text (without counting the figures and the supp. material)

2. Structure: 

    - Intro (1 paragraph) explicando el contexto biologico (que hace su factor,) ....
    - Synthetic Mat & Met
    - Results
    - Conclusions and discussion
    - Suppl. Material
    
          - A sufficient description of all the parameters of each step of your analysis to enable someone else to reproduce it. Can be the full command line with all the attributes (no need to disdcuss details)
          

