<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2019-02-08" />

<title>Practical: analysis of ChIP-seq peaks</title>

<script src="chip-seq_analysis_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="chip-seq_analysis_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="chip-seq_analysis_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="chip-seq_analysis_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="chip-seq_analysis_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="chip-seq_analysis_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="chip-seq_analysis_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="chip-seq_analysis_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="chip-seq_analysis_files/navigation-1.1/tabsets.js"></script>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Practical: analysis of ChIP-seq peaks</h1>
<h3 class="subtitle"><em>LCG_BEII 2019</em></h3>
<h4 class="author"><em>Jacques van Helden</em></h4>
<h4 class="date"><em>2019-02-08</em></h4>

</div>


<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this practrical, we will combine various bioinformatics resources to extract information from ChIP-seq experiments, and evaluate the quality of the peaks.</p>
</div>
<div id="resources" class="section level2">
<h2>Resources</h2>
<table style="width:90%;">
<colgroup>
<col width="36%" />
<col width="54%" />
</colgroup>
<thead>
<tr class="header">
<th>Resource Name</th>
<th>URL</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ReMap</td>
<td>A database of ChIP-seq peaks (<a href="http://remap.cisreg.eu/" class="uri">http://remap.cisreg.eu/</a>)</td>
</tr>
<tr class="even">
<td>Jaspar</td>
<td>A database of eukaryote TF binding motifs, mainly built from CHIP-seq peaks (<a href="http://jaspar.genereg.net/" class="uri">http://jaspar.genereg.net/</a>)</td>
</tr>
<tr class="odd">
<td>Hocomoco</td>
<td>A database of Human TF binding motifs, mainly built from CHIP-seq peaks (<a href="http://hocomoco11.autosome.ru/" class="uri">http://hocomoco11.autosome.ru/</a>)</td>
</tr>
<tr class="even">
<td>RSAT</td>
<td>Regulatory Sequence Analysis Tools <a href="http://rsat.eu/" class="uri">http://rsat.eu/</a></td>
</tr>
<tr class="odd">
<td>RSAT Metazoa</td>
<td><a href="http://metazoa.rsat.eu/" class="uri">http://metazoa.rsat.eu/</a></td>
</tr>
</tbody>
</table>
</div>
<div id="creating-a-workspace-for-this-practical" class="section level2">
<h2>Creating a workspace for this practical</h2>
<ul>
<li>Create a new folder to store the results of this practical (e.g. <code>~/LCG_BEII/chip-seq_practical/</code>)</li>
<li>Set this folder as your workiing direct</li>
</ul>
</div>
<div id="getting-peaks-from-remap" class="section level2">
<h2>Getting peaks from ReMap</h2>
<p>ReMap (<a href="http://remap.cisreg.eu/" class="uri">http://remap.cisreg.eu/</a>) is a database of TF binding regions based on an extensive re-analysis of published ChIP-seq data for human transcription factors. ory (<code>setwd()</code>)</p>
<ul>
<li>Open a connection to ReMap</li>
<li>Explore the interface</li>
<li>Choose a transcription factor and a tissue / cell type of interest (for this tutorial I will use Sox2 in Embryonic stem cells, ESC)</li>
<li>Get the peak coordinates for your TF in the selected tissue / cell type</li>
</ul>
</div>
<div id="getting-motifs-from-reference-databases" class="section level2">
<h2>Getting motifs from reference databases</h2>
<ul>
<li><p>Open a connection to Jaspar</p>
<ul>
<li><p>Explore the database functionalities</p></li>
<li><p>Find the transcription factor binding motifs for your TF of interest. Download the corresponding matrix in your working directory.</p></li>
</ul></li>
<li><p>Do the same with the <a href="http://hocomoco11.autosome.ru/">Hocomoco</a> database.</p>
<ul>
<li>Note the classification of transcription factors on the home page. You can browse it to observe TF families, defined based on their binding domains.</li>
<li>Find the transcripiton factor of interest and save its binding motif in your work directory.</li>
</ul></li>
</ul>
</div>
<div id="discovering-motifs-with-rsat-peak-motifs" class="section level2">
<h2>Discovering motifs with RSAT peak-motifs</h2>
<ul>
<li><p>Open a connection to <a href="http://metazoa.rsat.eu/">RSAT Metazoa</a></p></li>
<li><p>Use the tool <strong>fetch-sequences</strong> to get the sequences of your peaks</p></li>
<li><p>Send the results to <strong>peak-motifs</strong></p>
<ul>
<li>as reference databases, select both Jaspar non-redundant vertebrates and Hocomoco</li>
<li>leave all other parameters unchanged an run the analyses</li>
</ul></li>
<li><p>Interpret the results:</p>
<ul>
<li>Did you obtain significant results? How significant ?</li>
<li>If so, were they characterized by over-representation (<em>oligo-analysis</em>), positional bias (<em>position-analysis</em>) or both?</li>
<li>Did some of the discovered motifs match the reference motifs for your TF in Jaspar and/or Hocomoco ?</li>
<li>Did you discover other motifs</li>
<li></li>
</ul></li>
</ul>
</div>
<div id="negative-control" class="section level2">
<h2>Negative control</h2>
<ul>
<li><p>Use <strong>random genome fragments</strong> to pick up random regions having the same sizes as your peaks</p></li>
<li><p>Run the same analyses as above with peak-motifs</p></li>
<li><p>Compare the results of this negative control with those obtained above with your peaks, and interpret them.</p></li>
</ul>
</div>
<div id="motif-encrihment-analysis" class="section level2">
<h2>Motif encrihment analysis</h2>
<ul>
<li><p>On the RSAT server, open the tool<strong>matrix-quality</strong></p></li>
<li><p>Submit your peak sequences</p></li>
<li><p>Enter the JASPAR and Hocomoco reference motifs</p></li>
<li><p>Run the analyses</p></li>
<li><p>Interpret the results</p>
<ul>
<li>Do you observe an enrichment of your peaks for Hocomoco and/or Jaspar reference motifs?</li>
</ul></li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
