---
title: "Practical: analysis of ChIP-seq peaks"
author: "Jacques van Helden"
date: '`r Sys.Date()`'
output:
  html_document:
    self_contained: no
    fig_caption: yes
    highlight: zenburn
    theme: cerulean
    toc: yes
    toc_depth: 3
    toc_float: yes
  ioslides_presentation:
    slide_level: 2
    self_contained: no
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    smaller: yes
    toc: yes
    widescreen: yes
  revealjs::revealjs_presentation:
    theme: night
    transition: none
    self_contained: true
    css: ../../slides.css
  slidy_presentation:
    smart: no
    slide_level: 2
    self_contained: yes
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    highlight: tango
    incremental: no
    keep_md: yes
    smaller: yes
    theme: cerulean
    toc: yes
    widescreen: yes
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
  beamer_presentation:
    colortheme: dolphin
    fig_caption: yes
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    highlight: tango
    incremental: no
    keep_tex: no
    slide_level: 2
    theme: Montpellier
    toc: yes
font-import: http://fonts.googleapis.com/css?family=Risque
subtitle: LCG_BEII 2019
font-family: Garamond
transition: linear
---

```{r include=FALSE, echo=FALSE, eval=TRUE}
library(knitr)
library(kableExtra)
# library(formattable)

options(width = 300)
# options(encoding = 'UTF-8')
knitr::opts_chunk$set(
  fig.width = 7, fig.height = 5, 
  fig.path = 'figures/R_intro_',
  fig.align = "center", 
  size = "tiny", 
  echo = TRUE, eval = TRUE, 
  warning = FALSE, message = FALSE, 
  results = FALSE, comment = "")

options(scipen = 12) ## Max number of digits for non-scientific notation
# knitr::asis_output("\\footnotesize")

```


## Introduction

In this practrical, we will combine various bioinformatics resources to extract information from ChIP-seq experiments, and evaluate the quality of the peaks.

## Resources


| Resource Name | URL |
|-------------------------|--------------------------------------|
| ReMap | A database of ChIP-seq peaks (<http://remap.cisreg.eu/>) |
| Jaspar | A database of eukaryote TF binding motifs, mainly built from CHIP-seq peaks (<>) |
| Hocomoco | A database of Human TF binding motifs, mainly built from CHIP-seq peaks (<>) |
| RSAT| Regulatory Sequence Analysis Tools <http://rsat.eu/> |
| RSAT Metazoa | <http://metazoa.rsat.eu/> |

## Creating a workspace for this practical

- Create a new folder to store the results of this practical (e.g.  `~/LCG_BEII/chip-seq_practical/`)
- Set this folder as your workiing direct

## Getting peaks from ReMap

ReMap (<http://remap.cisreg.eu/>) is a database of TF binding regions based on an extensive re-analysis of published ChIP-seq data for human transcription factors. ory (`setwd()`)

- Open a connection to ReMap
- Explore the interface
- Choose a transcription factor and a tissue / cell type of interest (for this tutorial I will use Sox2 in Embryonic stem cells, ESC) 
- Get the peak coordinates for your TF in the selected tissue / cell type

## Getting motifs from reference databases

- OPen a connection to Jaspar, and find the transcription factor binding motifs for your TF of interest. Download the corresponding matrix in your working directory.

- Do the same with Hocomoco




